courier/template/verification_invalid_test.go

```diff
@@ -1,24 +1,77 @@
 package template_test
 
 import (
-	"testing"
-
-	"github.com/stretchr/testify/assert"
+	"encoding/base64"
+	"github.com/julienschmidt/httprouter"
+	"github.com/ory/kratos/courier/template/testhelpers"
 	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
 
 	"github.com/ory/kratos/courier/template"
 	"github.com/ory/kratos/internal"
 )
 
 func TestVerifyInvalid(t *testing.T) {
-	conf, _ := internal.NewFastRegistryWithMocks(t)
-	tpl := template.NewVerificationInvalid(conf, &template.VerificationInvalidModel{})
 
-	rendered, err := tpl.EmailBody()
-	require.NoError(t, err)
-	assert.NotEmpty(t, rendered)
+	t.Run("test=with courier templates directory", func(t *testing.T) {
+		conf, _ := internal.NewFastRegistryWithMocks(t)
+		tpl := template.NewVerificationInvalid(conf, &template.VerificationInvalidModel{})
+
+		testhelpers.TestRendered(t, tpl)
+	})
+
+	t.Run("test=with remote resources", func(t *testing.T) {
```

Comment:
This test structure is present 4 times, can we DRY it up into a common helper function or is there a good reason to have it duplicated?

Reply:
fixed :)